node/src/components/storage/blob_cache.rs

```diff
@@ -0,0 +1,167 @@
+//! A reference cache for items.
+
+use std::{
+    collections::HashMap,
+    hash::Hash,
+    sync::{Arc, Weak},
+};
+
+use datasize::DataSize;
+
+/// A cache of serialized items.
+///
+/// Maintains a collection of weak references and automatically purges them in configurable
+/// intervals.
+#[derive(DataSize, Debug)]
+pub(crate) struct BlobCache<I> {
```

Comment:
```suggestion
pub(super) struct BlobCache<I> {
```

Same change for all `pub(crate)` in this module.